tsdb/index/tsi1/tag_block.go

```diff
@@ -316,16 +316,17 @@ func (e *TagBlockValueElem) Value() []byte { return e.value }
 // SeriesN returns the series count.
 func (e *TagBlockValueElem) SeriesN() uint64 { return e.series.n }
 
-// SeriesID returns series ID at an index.
-func (e *TagBlockValueElem) SeriesID(i int) uint64 {
-	return binary.BigEndian.Uint64(e.series.data[i*SeriesIDSize:])
-}
-
 // SeriesIDs returns a list decoded series ids.
 func (e *TagBlockValueElem) SeriesIDs() []uint64 {
-	a := make([]uint64, e.series.n)
-	for i := 0; i < int(e.series.n); i++ {
-		a[i] = e.SeriesID(i)
+	a := make([]uint64, 0, e.series.n)
```

Comment:
Same question regarding memory size when we have a lot of series sharing the same tag value. Could we end up with huge allocations here?